projects/packages/publicize/src/class-publicize-base.php

```diff
@@ -527,6 +527,23 @@ public function get_display_name( $service_name, $connection ) {
 		return $connection_display;
 	}
 
+	/**
+	 * Returns the account name for the Connection. For services like Instagram, we need both the username and the account's name.
+	 *
+	 * @param string       $service_name 'facebook', 'linkedin', etc.
+	 * @param object|array $connection The Connection object (WordPress.com) or array (Jetpack).
+	 * @return string
+	 */
+	public function get_username( $service_name, $connection ) {
+		$cmeta = $this->get_connection_meta( $connection );
+
+		if ( 'instagram-business' === $service_name && isset( $cmeta['connection_data']['meta']['username'] ) ) {
```

Comment:
Shall we make this generic? If the `username` is set then we use that and otherwise fallback to the display name?